### DISCUSSION TITLE
wallpaper changer seems to not respect image filter

### DISCUSSION THREAD

**Discussion by MaKraMc** [you-apps/WallYou#171] [General]
### Steps to reproduce

Example: Unsplash
1. Navigate to Unsplash, set a filter for "rare" images e.g. "giraffe". The app now displays only giraffe pictures.
2. Enable Wallpaper changer and set an interval.
3. Wait for the wallpaper changer to automatically change the wallpaper.

### Expected behavior

The wallpaper is set to a picture of a giraffe, as configured in the filter.

### Actual behavior

Wallpaper is set to some random picture from unsplash, ignoring the set filter.

### Wall You version

6.2 (23)

### Android version

Android 14 (Graphene OS)

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
For some wallpaper sources, the filters are respected, for some they're not.

That's simply due to the fact that some APIs allow receiving a random wallpaper by just doing one API call, while for other APIs, we have to do a normal request to fetch a list of wallpapers and then have to choose a random one ourself.

If using the second method, the wallpapers will respect the filters you have set. However, they might be starting to repeat really quick because the random wallpapers are only being extracted from a list of a few wallpapers which might always be the same. The other method is really 100% random since it chooses random wallpapers at API level.

Both of the approaches have (dis)advantages, but I think it's not a good idea to change the sources currently using method 2 to using method 1.

Opinions @M00NJ @SuhasDissa ?

**Comment by M00NJ** [comment]
I agree with Bnyro on this.
